### DISCUSSION TITLE
v2.0 : Error processing request: No vector store provided. Please provide a vector store to train the agent.

### DISCUSSION THREAD

**Discussion by Hrishikesh-Dutta0078** [sinaptik-ai/pandas-ai#973] [Q&A]
I wanted to test the training feature in v2.0 but I am getting the following error :
Error processing request: No vector store provided. Please provide a vector store to train the agent.
I am using AzureOpenAI as the llm. Here is the code :
df = Agent(data_frames, config={
            "llm": llm,
            "enable_cache": False,
            "enforce_privacy": True,
            "open_charts": False,
            "response_parser": PandasDataFrame
        })
        df.train(docs = "boq attainment percent is calculated as 'Total Boq-Actual/Total Boq-Plan'")
        
        It works with OpenAI api key.

**Comment by junle-chen** [comment]
I encountered this problem when training, and I used openai api key.
